### PR TITLE
Update binder gen emitted-interceptor nullability to match framework impl

### PR DESCRIPTION
**Pull request by layomia**
Addresses https://github.com/dotnet/runtime/issues/90987. RC 2 candidate.

### CODE REVIEW

**Review by eerhardt — APPROVED** [review]
LGTM

File: src/libraries/Microsoft.Extensions.Configuration.Binder/src/ConfigurationBinder.cs [hunk 1]
```diff
@@ -305,125 +312,122 @@ private static void BindProperty(PropertyInfo property, object instance, IConfig
                 return;
             }
 
-            if (config != null)
+            if (config.GetChildren().Any())
```
Comment (tarekgh) [inline src/libraries/Microsoft.Extensions.Configuration.Binder/src/ConfigurationBinder.cs:315]: >config [](http://example.com/codeflow?start=16&length=6)

are we sure we never get `config` with null values? if so can you assert for that?

**Review by tarekgh — APPROVED** [review]

File: src/libraries/Microsoft.Extensions.Configuration.Binder/src/ConfigurationBinder.cs (hunk 1 above)
Comment (eerhardt) [reply src/libraries/Microsoft.Extensions.Configuration.Binder/src/ConfigurationBinder.cs:315]: What's the value of the assert when the method has it annotated as "does not accept `null`"?

**Comment by layomia** [comment]
Yeah, I'm going to backport it.

**Pull request merged**